lib/zipkin-tracer/trace_client.rb

```diff
@@ -0,0 +1,26 @@
+module ZipkinTracer
+  class TraceClient
+    LOCAL_COMPONENT = 'lc'.freeze
+    STRING_TYPE = 'STRING'.freeze
+
+    def self.record(key)
+      Trace.record(Trace::Annotation.new(key, Trace.default_endpoint))
+    end
+
+    def self.record_tag(key, value)
+      Trace.record(Trace::BinaryAnnotation.new(key, value, STRING_TYPE, Trace.default_endpoint))
+    end
+
+    def self.record_local_component(value)
+      record_tag(LOCAL_COMPONENT, value)
+    end
+
+    def self.trace(key = nil)
```

Comment:
I think this is a very good idea and I like it very much but then you should be creating new spans instead.
It will kind of give you the start/end in a more meaninful way and it will also show beautiful in the zipkin server.

You will need to check how to create new spans, maybe it's not easy as I think that's inside thrift code but very worth it